syft/lib.go

```diff
@@ -61,24 +64,54 @@ func IdentifyDistro(s scope.Scope) distro.Distro {
 	return d
 }
 
-// Catalog the given scope, which may represent a container image or filesystem. Returns the discovered set of packages.
-func CatalogFromScope(s scope.Scope) (*pkg.Catalog, error) {
+// Catalog the given source, which may represent a container image or filesystem. Returns the discovered set of packages.
+func CatalogFromScope(s source.Source) (*pkg.Catalog, error) {
 	log.Info("building the catalog")
 
 	// conditionally have two sets of catalogers
 	var catalogers []cataloger.Cataloger
-	switch s.Scheme {
-	case scope.ImageScheme:
+	switch s.Metadata.Scheme {
+	case source.ImageScheme:
 		catalogers = cataloger.ImageCatalogers()
-	case scope.DirectoryScheme:
+	case source.DirectoryScheme:
 		catalogers = cataloger.DirectoryCatalogers()
 	default:
-		return nil, fmt.Errorf("unable to determine cataloger set from scheme=%+v", s.Scheme)
+		return nil, fmt.Errorf("unable to determine cataloger set from scheme=%+v", s.Metadata.Scheme)
 	}
 
 	return cataloger.Catalog(s.Resolver, catalogers...)
 }
 
+// CatalogFromJSON takes an existing syft report and generates catalog primitives.
+func CatalogFromJSON(reader io.Reader) (*pkg.Catalog, *distro.Distro, source.Metadata, error) {
```

Comment:
Nit: It's not obvious why `source.Metadata` is returned without a pointer, but the same is not true for `*distro.Distro`. Both seem like reasonably lightweight structs, so I might suggest making `*distro.Distro` not a pointer type, and returning the zero-value struct on the "error" return paths (just like what's done for `source.Metadata{}`).

Reply:
I'll minimize the pointer usage (and while I'm at it order the return values consistently)